renderer/src/common/components/settings/registry/schema.ts

```diff
@@ -0,0 +1,56 @@
+import { z } from 'zod/v4'
+
+export const registryFormSchema = z
+  .object({
+    type: z.enum(['file', 'url', 'default']).default('default'),
+    source: z.string().optional(),
+  })
+  .refine(
+    (data) => {
+      if (data.type === 'url' || data.type === 'file') {
+        return data.source && data.source.trim().length > 0
+      }
+      return true
+    },
+    {
+      message: 'Registry URL or file path is required',
+      path: ['source'],
+    }
+  )
+  .refine(
+    (data) => {
+      if (data.type === 'url' || data.type === 'file') {
+        return data.source?.endsWith('.json')
+      }
+      return true
+    },
+    {
+      message: 'Registry must be a .json file',
+      path: ['source'],
+    }
+  )
+  .refine(
+    (data) => {
+      if (data.type === 'url' && data.source) {
+        if (
+          !data.source.startsWith('http://') &&
```

Comment:
It needs to be https or it will fail when trying to fetch the registry. This is a bug in the backend that we don't validate it.